src/mapnik_vector_tile.cpp

```diff
@@ -1688,89 +1605,94 @@ NAN_METHOD(VectorTile::toJSON)
+                feature_obj->Set(NanNew("geometry"),g_arr);
+                Local<Object> att_obj = NanNew<Object>();
+                for (int m = 0; m < f.tags_size(); m += 2)
+                {
+                    std::size_t key_name = f.tags(m);
+                    std::size_t key_value = f.tags(m + 1);
+                    if (key_name < static_cast<std::size_t>(layer.keys_size())
+                        && key_value < static_cast<std::size_t>(layer.values_size()))
                     {
-                        att_obj->Set(NanNew(name.c_str()), NanUndefined());
+                        std::string const& name = layer.keys(key_name);
+                        vector_tile::Tile_Value const& value = layer.values(key_value);
+                        if (value.has_string_value())
+                        {
+                            att_obj->Set(NanNew(name.c_str()), NanNew(value.string_value().c_str()));
+                        }
+                        else if (value.has_int_value())
+                        {
+                            att_obj->Set(NanNew(name.c_str()), NanNew<Number>(value.int_value()));
+                        }
+                        else if (value.has_double_value())
+                        {
+                            att_obj->Set(NanNew(name.c_str()), NanNew<Number>(value.double_value()));
+                        }
+                        else if (value.has_float_value())
+                        {
+                            att_obj->Set(NanNew(name.c_str()), NanNew<Number>(value.float_value()));
+                        }
+                        else if (value.has_bool_value())
+                        {
+                            att_obj->Set(NanNew(name.c_str()), NanNew<Boolean>(value.bool_value()));
+                        }
+                        else if (value.has_sint_value())
+                        {
+                            att_obj->Set(NanNew(name.c_str()), NanNew<Number>(value.sint_value()));
+                        }
+                        else if (value.has_uint_value())
+                        {
+                            att_obj->Set(NanNew(name.c_str()), NanNew<Number>(value.uint_value()));
+                        }
+                        else
+                        {
+                            att_obj->Set(NanNew(name.c_str()), NanUndefined());
+                        }
                     }
+                    feature_obj->Set(NanNew("properties"),att_obj);
                 }
-                feature_obj->Set(NanNew("properties"),att_obj);
-            }
 
-            f_arr->Set(j,feature_obj);
+                f_arr->Set(j,feature_obj);
+            }
+            layer_obj->Set(NanNew("features"), f_arr);
+            arr->Set(i, layer_obj);
         }
-        layer_obj->Set(NanNew("features"), f_arr);
-        arr->Set(i, layer_obj);
+        NanReturnValue(arr);
+    } catch (std::exception const& ex) {
+        NanThrowError(ex.what());
```

Comment:
No test of exception being caught here, lets try to confirm we need it since try was added with a test?